module-2/lab-pivot-table-and-correlation/your_code/main.ipynb

```diff
@@ -21,13 +21,13 @@
   },
```

Comment:
- To the question Use matplotlib to visually plot Minutes Sedentary vs Steps. Based on the results of the computed correlation and the plot. What can you say about these 2 variables?, your comment "Both variables are independent they have a negative correlation, the increase of one of them doesnt have any effect on the other one", you better say there is no relationship between the minutes sedentary and number of steps what is proved by a near 0 corr coef. 
- Based on the result of the correlation matrix. What are the highly correlated features with the variable Steps?: You only need to order the Steps column from your correlation matrix, not the entire matrix. 
- To the question Use matplotlib to visually plot Study time vs Grade. Based on the results of the computed correlation and the plot. What can you say about these 2 variables?, your comment "There is a low positive correlation between the study time and the grade" is not correct. The corr coef, both pearson and spearman are above 0.75 which is not a low correlation. 
- To the question Could you explain the difference between Pearson and Spearman correlation?, your comment "Spearman correlation measures relationships between variables that are not necessarily linear.
####Spearman vs Pearson: Pearson correlation measures relationships between variables that are linear" is not quite correct. 
Spearman vs Pearson: It is usually a little bit higher Spearman than Pearson but it doesn't mean necessarily that the relationship is not linear. Why? The Pearson is most appropriate for measurements taken from an interval scale (temperature, dates, lengths, etc), while the Spearman is best for measurements taken from ordinal scales (rank orders, spectrum of values (agree, neutral, disagree), or healthy vs non-healthy). However, with large samples, the correlation coefficients will be similar. The largest differences will be seen with small sample sizes. Pearson's assumes constant variance and linearity. If your variable violates that, Spearman's would be the best to use. This is the best explanation I have seen up to now (extract from quora). In other words, Spearman's coefficient measures the rank order of the points (it doesn't care where the points are) while Pearson measures the linear relationship between the two. In the end, it's not about a better or worse correlation, it's more about that we should use one or another depending on the characteristics of our variables.
In statistics and generally speaking, the fact two variables correlate cannot prove anything - only further research can actually prove that one thing affects the other.Data reliability is related to the size of the sample. The more data you collect, the more reliable your result. ####